centreon/src/Core/Common/Infrastructure/Repository/FileDataStoreEngine.php

```diff
@@ -0,0 +1,277 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types = 1);
+
+namespace Core\Common\Infrastructure\Repository;
+
+use Centreon\Domain\Repository\Interfaces\DataStorageEngineInterface;
+use Security\Encryption;
+
+class FileDataStoreEngine implements DataStorageEngineInterface
+{
+    private bool $inTransaction = false;
+
+    private string $workingDirectory = '';
+
+    private string $lastError = '';
+
+    /**
+     * @param string $absoluteMediaPath Absolute destination path for the media
+     * @param bool $throwsException Indicates whether an exception should be thrown in the event of an error,
+     *                              false by default
+     *
+     * @throws \Exception
+     */
+    public function __construct(private string $absoluteMediaPath, private bool $throwsException = false)
+    {
+        $this->absoluteMediaPath = realpath($absoluteMediaPath)
+            ?: throw new \Exception(sprintf('Path invalid \'%s\'', $absoluteMediaPath));
+    }
+
+    /**
+     * Defines whether an exception should be thrown in the event of an error.
+     *
+     * @param bool $throwsException
+     */
+    public function throwsException(bool $throwsException): void
+    {
+        $this->throwsException = $throwsException;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function rollbackTransaction(): bool
+    {
+        try {
+            $rollbackStatus = $this->rollbackFileSystem();
+            $this->inTransaction = false;
+
+            return $rollbackStatus;
+        } catch (\Throwable $ex) {
+            $this->lastError = $ex->getMessage();
+            if ($this->throwsException) {
+                throw $ex;
+            }
+
+            return false;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function startTransaction(): bool
+    {
+        if ($this->inTransaction) {
+            return false;
+        }
+        $tempDirectory = str_replace(DIRECTORY_SEPARATOR, '-', Encryption::generateRandomString(32));
+        $this->workingDirectory = sys_get_temp_dir() . DIRECTORY_SEPARATOR . $tempDirectory;
+        if (! mkdir($this->workingDirectory)) {
+            throw new \Exception(
+                sprintf('Unable to create a directory \'%s\'', $this->workingDirectory)
+            );
+        }
+        $this->inTransaction = true;
+
+        return true;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function commitTransaction(): bool
+    {
+        $this->inTransaction = false;
+
+        return $this->commitFileSystem();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function isAlreadyinTransaction(): bool
+    {
+        return $this->inTransaction;
+    }
+
+    /**
+     * @param string $pathname
+     * @param string $data
+     * @param int $flags
+     *
+     * @throws \Exception
+     *
+     * @return int|bool
```

Comment:
100% accurate phpdoc for phpstan is recommendend because the `false` type is supported in php 8.2+
```suggestion
     * @return int|false
```